src/librustc_codegen_llvm/back/link.rs

```diff
@@ -221,7 +219,8 @@ fn preserve_objects_for_their_debuginfo(sess: &Session) -> bool {
 
 pub(crate) fn each_linked_rlib(sess: &Session,
                                info: &CrateInfo,
-                               f: &mut dyn FnMut(CrateNum, &Path)) -> Result<(), String> {
+                               f: &mut dyn FnMut(CrateNum, &Path)) -> Result<(), Cow<'static, str>>
```

Comment:
Same as before -- I'd rather we make this an error enum if we're making performance optimizations here since that's going to be more performant in all likelihood

Reply:
I guess I'll just revert this one, most of the errors are `String`s and I wouldn't want to complicate this PR further.